test/unittests/libevm/VMTest.cpp

```diff
@@ -694,6 +695,88 @@ class AlethInterpreterBalanceFixture : public BalanceFixture
 public:
     AlethInterpreterBalanceFixture() : BalanceFixture{new EVMC{evmc_create_interpreter()}} {}
 };
+
+class CallSelfFixture : public TestOutputHelperFixture
+{
+public:
+    explicit CallSelfFixture(VMFace* _vm) : vm{_vm} { state.addBalance(address, 1 * ether); }
+
+    void testCallSelf()
+    {
+        constexpr size_t codeElementCount = 4;
+        bytes codeArray[] = {
+            // staticcall must come first (we need to pass if the code performs a
```

Comment:
well you could also just use `i == staticCallCaseIndex` as a condition there in the last argument of `ExtVM` constructor